test/src/java/org/jdom2/test/cases/input/TestStAXStreamBuilder.java

```diff
@@ -113,16 +116,39 @@ public void testDocTypeSimpleDocument() {
 	public void testComplexDocument() {
 		checkStAX("/DOMBuilder/complex.xml", false);
 	}
-	
+
+	@Test
+	public void testReportPrologWsDocument() throws Exception {
+		URLClassLoader classLoader = new URLClassLoader(
+				new URL[] { new File("lib/woodstox-core-5.0.2.jar").toURI().toURL(),
+							new File("lib/stax2-api-3.1.4.jar").toURI().toURL() });
+		
+		checkStAX("/DOMBuilder/report_prolog_ws.xml", false, "javax.xml.stream.XMLInputFactory", classLoader, 5);

```

Comment:
Woodstox has a provider ( META-INF/services/javax.xml.stream.XMLInputFactory ) to define its implementation of the XmlInputFactory.